### PR TITLE
Document supported cable-drivers

### PR DESCRIPTION
**Pull request by nyechiel**
Fixes: https://github.com/submariner-io/submariner-website/issues/146
Signed-off-by: nyechiel <n.yechiel@gmail.com>

### CODE REVIEW

**Review by dfarrell07 — APPROVED** [review]

File: src/content/deployment/subctl/_index.en.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ deployment.
 <!-- markdownlint-disable line-length -->
 | Flag                               | Description
 |:-----------------------------------|:----------------------------------------------------------------------------|
-| `--cable-driver` `<string>`        | Cable driver implementation (defaults to strongSwan IPsec)
+| `--cable-driver` `<string>`        | Cable driver implementation. Available options are `strongswan` (default) and `libreswan`
```
Comment (tpantelis) [inline src/content/deployment/subctl/_index.en.md:98]:
```suggestion
| `--cable-driver` `<string>`        | Cable driver implementation. Available options are `strongswan` (default), `libreswan` and `wireguard`.
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: src/content/deployment/subctl/_index.en.md (hunk 1 above)
Comment (dfarrell07) [reply src/content/deployment/subctl/_index.en.md:98]: Good suggestion, but let's make sure we add it as a different commit or update the commit message of the proposed commit. I'd also recommend a serial comma.

Comment (nyechiel) [reply src/content/deployment/subctl/_index.en.md:98]: I thought about this while submitting the change. Are we OK with documenting WireGuard in here? Personally, I don't have much experience with it and I am not sure if there are other prerequisites we need to document /cc @mangelajo

Comment (dfarrell07) [reply src/content/deployment/subctl/_index.en.md:98]: There are other prerequisites for WireGuard, see: https://github.com/submariner-io/submariner/blob/master/.github/workflows/e2e.yml#L31

Comment (nyechiel) [reply src/content/deployment/subctl/_index.en.md:98]: OK, let me change the commit message and PR description. I will also submit a separate PR for the WireGuard prerequisites.

Comment (tpantelis) [reply src/content/deployment/subctl/_index.en.md]: We're wording it as "_Available options are..._" which implies all of them so I think wireguard needs to be included. We already document all of them elsewhere so perhaps we should word it as: "_Available options include `strongswan` (default) and `libreswan`. See the complete list  of options [here](../architecture/gateway-engine)_".

File: src/content/architecture/gateway-engine/_index.en.md [hunk 2]
```diff
@@ -13,7 +13,14 @@ that maintains the tunnels. The following implementations are available:
...
   this is currently the default;
 * an IPsec implementation using [Libreswan](https://libreswan.org/);
-* an implementation for [WireGuard](https://www.wireguard.com/).
+* an implementation for [WireGuard](https://www.wireguard.com/) (via the [wgctrl](https://github.com/WireGuard/wgctrl-go) library).
+
+The cable driver can be specified via the `--cable-driver` flag while joining a cluster using `subctl`. For more information, please refer
+to the [`subctl` guide](../../deployment/subctl/).
+
+{{% notice note %}}
+WireGuard needs to be installed on Gateway nodes. For installation instructions see [here](https://www.wireguard.com/install/).
```
Comment (dfarrell07) [inline src/content/architecture/gateway-engine/_index.en.md:22]:
```suggestion
WireGuard needs to be installed on Gateway nodes. See the [WireGuard installation instructions](https://www.wireguard.com/install/).
```

File: src/content/deployment/subctl/_index.en.md [hunk 3]
```diff
@@ -95,7 +95,7 @@ deployment.
 <!-- markdownlint-disable line-length -->
 | Flag                               | Description
 |:-----------------------------------|:----------------------------------------------------------------------------|
-| `--cable-driver` `<string>`        | Cable driver implementation (defaults to strongSwan IPsec)
+| `--cable-driver` `<string>`        | Cable driver implementation; available options are `strongswan` (default), `libreswan` and `wireguard`
```
Comment (dfarrell07) [inline src/content/deployment/subctl/_index.en.md:98]:
```suggestion
| `--cable-driver` `<string>`        | Cable driver implementation; available options are `strongswan` (default), `libreswan`, and `wireguard`
```

**Review by dfarrell07 — APPROVED** [review]
LGTM, I'd suggest we squash during merge.

File: src/content/architecture/gateway-engine/_index.en.md [hunk 4]
```diff
@@ -13,7 +13,14 @@ that maintains the tunnels. The following implementations are available:
   [goStrongswanVici](https://github.com/bronze1man/goStrongswanVici) library);
   this is currently the default;
 * an IPsec implementation using [Libreswan](https://libreswan.org/);
```
Comment (tpantelis) [inline src/content/architecture/gateway-engine/_index.en.md:15]:
```suggestion
* an IPsec implementation using [Libreswan](https://libreswan.org/).
```

File: src/content/deployment/subctl/_index.en.md (hunk 3 above)
Comment (tpantelis) [inline src/content/deployment/subctl/_index.en.md]:
```suggestion
| `--cable-driver` `<string>`        | Cable driver implementation. The available options are `strongswan` (default), `libreswan` and `wireguard`.
```

Comment (nyechiel) [reply src/content/deployment/subctl/_index.en.md]: > 

None of the other sentences in the table include a period at the end. This is true for all tables in that doc. We need to be consistent

File: src/content/architecture/gateway-engine/_index.en.md [hunk 5]
```diff
@@ -10,10 +10,16 @@ The Gateway Engine has a pluggable architecture for the cable engine component
 that maintains the tunnels. The following implementations are available:
 
 * an IPsec implementation using [strongSwan](https://www.strongswan.org/) (via the
-  [goStrongswanVici](https://github.com/bronze1man/goStrongswanVici) library);
-  this is currently the default;
-* an IPsec implementation using [Libreswan](https://libreswan.org/);
-* an implementation for [WireGuard](https://www.wireguard.com/).
+  [goStrongswanVici](https://github.com/bronze1man/goStrongswanVici) library). This is currently the default;
```
Comment (tpantelis) [inline src/content/architecture/gateway-engine/_index.en.md:13]:
```suggestion
  [goStrongswanVici](https://github.com/bronze1man/goStrongswanVici) library). This is currently the default.
```

**Review by tpantelis — APPROVED** [review]

**Pull request merged**